### PR TITLE
feat(email): store outlook refresh token in db

### PR DESCRIPTION
**Pull request by whutchinson98**
We need to store outlook tokens in our db as they rotate on every use

### CODE REVIEW

File: services/authentication_service/src/microsoft_token_cipher.rs [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+            .ok_or(MicrosoftTokenCipherError::MalformedIdentity)?;
+        if fusionauth_user_id.is_empty() || fusionauth_user_id.contains('\0') {
+            return Err(MicrosoftTokenCipherError::MalformedIdentity);
+        }
+
+        Ok(Self {
+            fusionauth_user_id,
+            email_address,
+        })
+    }
```
[the remark this answers is not in the document]
Comment (whutchinson98) [reply services/authentication_service/src/microsoft_token_cipher.rs:306]: wouldn't the generic account email be the normalized email? stripped of + and lowercase?

File: services/authentication_service/src/api/oauth2/microsoft.rs [hunk 2]
```diff
@@ -38,6 +43,312 @@ struct MicrosoftLinkIdentity {
...
+                    )
+                })?;
+
+                match previous_refresh_token {
+                    Some(previous_refresh_token) => Ok(FusionLinkChange::Replaced {
+                        previous_refresh_token,
+                    }),
+                    None => Ok(FusionLinkChange::Unchanged),
+                }
+            }
```
[the remark this answers is not in the document]
Comment (whutchinson98) [reply services/authentication_service/src/api/oauth2/microsoft.rs:244]: Claim 1 — "reject an empty refresh_token before calling replace_identity_provider_grant": already handled in two places.
- The token exchange rejects empty refresh tokens before they ever reach link_identity: crates/fusionauth/src/microsoft/oauth.rs:154-158 returns an error if payload.refresh_token.trim().is_empty(), so MicrosoftExchangeTokenResponse.refresh_token is guaranteed non-empty in the production path.
- Even if an empty token somehow arrived, replace_identity_provider_grant itself no-ops on it: services/authentication_service/src/api/oauth2/account_link.rs:62-68 returns Ok(()) and explicitly leaves the existing grant untouched. Nothing gets clobbered, so a third guard in microsoft.rs:205-244 would be redundant.

Claim 2 — "replacement can succeed without a captured previous token, so Unchanged hides a reversible change": there is no such path. previous_refresh_token ends up None in microsoft.rs:216-221 in exactly three cases, and each one is a case where replace_identity_provider_grant provably changes nothing:
- Empty new token → replace no-ops (account_link.rs:62).
- No link matching the email's display_name → replace applies the same find predicate and skips (account_link.rs:75-84).
- Existing token equals the new token → replace returns early without unlink/relink (account_link.rs:86-88).

The capture condition in microsoft.rs mirrors the no-op conditions in the replace function one-for-one, so FusionLinkChange::Unchanged is accurate: whenever no previous token was captured, FusionAuth state was not modified and there is nothing for compensate_link to reverse. The case where a change does happen always yields Replaced { previous_refresh_token }, which compensate_link (microsoft.rs:328-338) restores — and the existing test database_failure_restores_replaced_link_and_cleans_pending_link covers exactly that.

Claim 3 — new LinkScenario variant and empty-token fixture in test.rs:282-285: moot, since it exists only to exercise the behavior changes from claims 1 and 2. Adding an empty-token fixture would test a state the exchange boundary makes unreachable.
